jetson/filter/src/__main__.py

```diff
@@ -283,16 +321,48 @@ def _runLKF(self):
         self.gps.fresh = False
         self.imu.fresh = False
 
+    def correct_bearing(self):
+        gps_bearing = self.gps.bearing.bearing_deg
+        imu_bearing = self.imu.bearing.bearing_deg
+
+        if time.time() - self.last_bearing_correction > self.config["Bearing_offset_interval_sec"]:
+            print("timeout passed")
+
+            if gps_bearing != 999:
+                print("valid gps bearing")
+
+                if self.nav_state == "Drive":
+                    print("drive state")
+
+                    if abs(self.drive_v_left - self.drive_v_right) < self.config["Drive_vel_diff_percent"]:
+                        print("driving straight")
+
+                        if abs(self.target_bearing - imu_bearing) < self.config["Bearing_error_threshold_deg"]:
```

Comment:
don't you want this to go the other way, when the error is too high correct?

Reply:
no because this isnt error between actual bearing and measured bearing, this is error between target bearing and measured bearing. So if we're driving straight, nav will always line up the rover so that the error between target and measured is close to zero, thats what our pid loop does. This is just another check to make sure we're driving straight and not turning, ig its kinda redundant now that I've added the drive velocity command check. The thing I *did* get backwards was the check on rate of target bearing change which I will fix

Reply:
oh right, my bad.